homeassistant/components/smlight/switch.py

```diff
@@ -86,22 +86,33 @@ def __init__(
 
         self._page, self._toggle = description.setting.value
 
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        self.async_on_remove(
+            self.coordinator.client.sse.register_settings_cb(
+                self.entity_description.setting, self.event_callback
+            )
+        )
+
     async def set_smlight(self, state: bool) -> None:
         """Set the state on SLZB device."""
         await self.coordinator.client.set_toggle(self._page, self._toggle, state)
 
+    @callback
+    def event_callback(self, event: SettingsEvent) -> None:
+        """Handle switch events from the SLZB device."""
+        if event.setting is not None:
```

Comment:
Duck type safety
```suggestion
        if isinstance(event.setting, dict):
```

Reply:
> Duck type safety

except here we know the exact type of the event, so should already be type safe?

Reply:
I have applied your other suggestions, Thanks!

Reply:
Entirely fair! Was mainly thinking along the lines of how we may know the type of event, but not the type of `event.settings` itself (hence the need for the `None` check at all)